rm-datasource/src/main/java/io/seata/rm/RMHandlerAT.java

```diff
@@ -40,43 +43,93 @@
 
     private static final int LIMIT_ROWS = 3000;
 
+    private final Map<String, Boolean> undoLogTableExistRecord = new ConcurrentHashMap<>();
+
     @Override
     public void handle(UndoLogDeleteRequest request) {
+        String resourceId = request.getResourceId();
         DataSourceManager dataSourceManager = (DataSourceManager)getResourceManager();
-        DataSourceProxy dataSourceProxy = dataSourceManager.get(request.getResourceId());
+        DataSourceProxy dataSourceProxy = dataSourceManager.get(resourceId);
         if (dataSourceProxy == null) {
-            LOGGER.warn("Failed to get dataSourceProxy for delete undolog on {}", request.getResourceId());
+            LOGGER.warn("Failed to get dataSourceProxy for delete undolog on {}", resourceId);
+            return;
+        }
+
+        boolean hasUndoLogTable = undoLogTableExistRecord.computeIfAbsent(resourceId, id -> checkUndoLogTableExist(dataSourceProxy));
+        if (!hasUndoLogTable) {
+            LOGGER.debug("resource({}) has no undo_log table, UndoLogDeleteRequest will be ignored", resourceId);
```

Comment:
if (logger.isDebugEnabled()) {
}

Reply:
Almost all Logger implement will do the same level check inside deug(), info()......
So we don't need to check again.

Reply:
ok